brave/src/test/java/com/linecorp/armeria/client/brave/BraveClientIntegrationTest.java

```diff
@@ -0,0 +1,224 @@
+import com.linecorp.armeria.client.ClientOptions;
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.NonWrappingRequestContext;
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.common.brave.RequestContextCurrentTraceContext;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.common.logging.RequestLogBuilder;
+import com.linecorp.armeria.common.metric.NoopMeterRegistry;
+
+import brave.SpanCustomizer;
+import brave.Tracing.Builder;
+import brave.http.HttpAdapter;
+import brave.http.HttpClientParser;
+import brave.propagation.CurrentTraceContext;
+import brave.propagation.StrictScopeDecorator;
+import brave.sampler.Sampler;
+import brave.test.http.ITHttpClient;
+import io.netty.channel.Channel;
+import io.netty.channel.EventLoop;
+import okhttp3.mockwebserver.MockResponse;
+import zipkin2.Span;
+
+public class BraveClientIntegrationTest extends ITHttpClient<HttpClient> {
+
+    // // Hide currentTraceContext in ITHttpClient
+    private CurrentTraceContext currentTraceContext =
+            RequestContextCurrentTraceContext.builder()
+                                             .addScopeDecorator(StrictScopeDecorator.create())
+                                             .build();
+
+    @Override
+    protected Builder tracingBuilder(Sampler sampler) {
+        return super.tracingBuilder(sampler).currentTraceContext(currentTraceContext);
+    }
+
+    @Override
+    protected HttpClient newClient(int port) {
+        return HttpClient.of("h1c://127.0.0.1:" + port, ClientOptions.of(
+                ClientOption.DECORATION.newValue(
+                        new ClientDecorationBuilder()
+                                .add(BraveClient.newDecorator(httpTracing))
+                                .build())));
+    }
+
+    @Override
+    @Test
+    public void makesChildOfCurrentSpan() throws Exception {
+        new DummyRequestContext().makeContextAware(() -> {
+            super.makesChildOfCurrentSpan();
+            return null;
+        }).call();
+    }
+
+    @Override
+    @Test
+    public void propagatesExtra_newTrace() throws Exception {
+        new DummyRequestContext().makeContextAware(() -> {
+            super.propagatesExtra_newTrace();
+            return null;
+        }).call();
+    }
+
+    @Override
+    @Test
+    public void propagatesExtra_unsampledTrace() throws Exception {
+        new DummyRequestContext().makeContextAware(() -> {
+            super.propagatesExtra_unsampledTrace();
+            return null;
+        }).call();
+    }
+
+    @Override
+    @Test
+    public void addsErrorTagOnTransportException() throws Exception {
+        throw new AssumptionViolatedException("Armeria won't record exception in this case.");
```

Comment:
@trustin curious if this is expected behavior.
Client got closed session exception, but we can't get it from RequestLog. Because it will call `onWrapperCompleted`. https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/client/Http1ResponseDecoder.java#L92

Reply:
I'm not 100% sure but I guess it's not an expected behavior. Do we have the same problem for HTTP/2? We could investigate this later though because it doesn't seem to belong to this pull request. Let's create a new issue for a follow-up.

Reply:
I switch to h2c in this PR and http/2 doesn't have this issue.

Reply:
https://github.com/line/armeria/issues/1914

Reply:
I would probably either make a base class and a top-level for each protocol to test, or a parameterized runner or similar as the behavior is different and important.